src/java_bytecode/jar_file.cpp

```diff
@@ -96,16 +138,17 @@ std::string jar_filet::get_entry(std::size_t i)
 
   std::string dest;
 
+  size_t real_index=filtered_jar[i];
```

Comment:
so the `index` vector is gone and `filtered_jart` is a `map<irep_idt, size_t>` now. Access to the files in the JAR is now via the file name.